### ISSUE TITLE
Allow modification and sharing of search results

### ISSUE THREAD

**Issue by zstadler** [IsraelHikingMap/Site#176]
Sharing is supposed to include everything that's seen on the map, but currently it does not include search results, such as places and trails.

In addition, there is no way to modify search results, such as changing a label on a POI or editing a trail.

I assume that if search results are editable, they will also be included in the sharing.

**Comment by HarelM** [comment]
I don't see this as neither a bug nor a high priority one.
I don't see a lot of value in allowing to change/edit/share it...
Please change it to low enhancement.

**Comment by zstadler** [comment]
I respectfully disagree. That's a main point in sharing.
Use case, I want to show you places to hike on the weekend. I search the site several times and send the share link. You respond by adding some places, deleting some, creating a route, and send a new share, which I further change.

**Comment by HarelM** [comment]
You are able to share files and links.
How many time did you share google search results?
Search is static, work is dynamic.
Search results are not part of the dynamic abilities of the site and I don't think it should change.
The only thing that might be worth looking into is the ability to take a search results and convert it to a route/POI, this way you'll be able to edit it and share it much like everything else.
Maybe a popup menu from the search results marker to allow the options to "create new route" or "add to <select route>", what do you think?

**Comment by zstadler** [comment]
I'm not looking for a new way to share the raw search results. Hyperlinks such as
&nbsp;&nbsp;[http://israelhiking.osm.org.il/#?q=עין נטף](http://israelhiking.osm.org.il/#?q=עין נטף)
are good enough.

The request is that the share link will preserve the specific items that were found using the search and selectively added to the map. IMO, this is not different than placing a waypoint or a polygon manually on the map.

The use case I have in mind is that I want to share with you the location of a place, such as "עין נטף". What I would do is:
- Search for the name
- Click on one of the two results I'm interested in
- Share the map 
- Anyone using the shared map sees exactly what I see - WYSISYG.

In other words, whenever a map with items from the search results is shared, I request that these items will be exported as additional routes (for ways and relations) and additional waypoints/POIs (for nodes).

**Comment by HarelM** [comment]
Sound like my idea should be able to cover what you need, wouldn't it?
> Maybe a popup menu from the search results marker to allow the options to "create new route" or "add to ", what do you think?

**Comment by zstadler** [comment]
I don't understand why the extra user action step is needed.
Moreover, it requires the user to decide to create something at search time, before the item is even shown on the map.

IMHO, if it's visible on the map, it should be included in the share URL, same as layers, routes and points. If something should not be shared, it need to be deleted or hidden.

Perhaps we can agree that search results will always be added as new points or routes so they can also be manipulated using the standard tools?
I believe it will remove the need to create special treatment for the search results when sharing.

**Comment by HarelM** [comment]
Consider the following user flow and let me know if it cover your requirements:
User searches for something.
Results appear on the screen.
User selects a relevant search item.
The search item appears on the map.
User is satisfied with the results.
User clicks the marker which will open a menu like all other markers has which will have the ability to "create route" or "remove".
Once "create route" is clicked a route will be created and the user will be able to edit it and share it like he would we other routes.

**Comment by zstadler** [comment]
That can work.

I don't see the value of the "create route" step.  Can you talk about the benefits of having it?

**Comment by HarelM** [comment]
The way I see it is based on the current search behavior, which is not aligned with the rest of the site mainly with the double click to remove the results.
My initial thought was as follows: in order for the search results to have the same feeling as other features it should have a pop-up menu with a trash bin so that the user will be able to know how to hide the search results.
Then I continued to thinking about how the user will search: if every search results the user clicks will be added to the route list a user will not want to search in order to prevent "route spam".
The equivalent I think is that you can search but not every time you click on link in the search results it becomes a bookmark.
So the same logic applies here, you can search, you can look at the results, but only add and manipulate things you really want.
Do you have an alternative user experience?

**Comment by zstadler** [comment]
I see your point and agree.
The user experience trade-off is between having an extra step to keep each search result and having a cluttered map with all the history of search results.
Repeated searches are used much more frequently than sharing the search results.

Your proposal also has a side-benefit of allowing the removal of search results on a touch interface that does not have a double-click, right!?

**Comment by HarelM** [comment]
I'm glad we agree, no reason not to implement it now :-) except it soon...
I do think every touch screen has a double click ability, I just think most applications do not use this option so mobile users do not think it's a possible option, but yes, this is a side benefit :-)

**Comment by zstadler** [comment]
Sometimes the search results window is too narrow:

<img width="212" alt="2016-09-22 13_15_43-israel hiking map" src="https://cloud.githubusercontent.com/assets/1304610/18744492/e4905ed4-80c6-11e6-9752-9a1f85bbbb69.png">

I was unable to find a consistent reproduction scenario.

**Comment by HarelM** [comment]
Happened to me for other pop-ups, related to synchronization between angular compiled html and leaflet pop-up presentation. I solved it with a delay of the pop-up presentation. I'll do it in this case as well.

**Comment by HarelM** [comment]
I looked at the code now and there's a delay of 300ms, I'm not sure there's something else I can do to make it work correctly, maybe set minimum width, I don't know...

**Comment by zstadler** [comment]
Is it possible to control the synchronization with an event?

**Comment by HarelM** [comment]
Not that I know of... :-(

**Comment by HarelM** [comment]
I can't manage to reproduce it anymore, please reopen if you see it again.
